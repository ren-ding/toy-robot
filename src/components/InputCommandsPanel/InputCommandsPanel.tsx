import React, { Component } from 'react';
import './styles/InputCommandsPanel.css'

interface Props{
  onInputCommands:(event: {target: { value: string }}) => void
}

const InputCommands = ({
  onInputCommands
}:Props)=>(
  <textarea rows={20} className='input-commands-textarea' onChange={onInputCommands}/>
);

export default class InputCommandsPanel extends Component<Props> {
  constructor(props:Props) {
    super(props);
  }

  render() {
    return (
      <div className='input-commands-panel-wrapper'>
        <span className='input-label'>Please input commands</span>
        <InputCommands 
          onInputCommands={this.props.onInputCommands}
        />
      </div>
    );
  }
}
