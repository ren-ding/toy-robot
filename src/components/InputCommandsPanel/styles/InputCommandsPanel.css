.input-commands-panel-wrapper
{
    flex: 1;
    margin: 20px
}
  
.input-commands-panel-wrapper .input-label
{
    font-size: 25px;
    display: block;
    padding: 5px;
}

.input-commands-panel-wrapper .input-commands-textarea
{
    font-size: 20px;
    width: 300px;
}
