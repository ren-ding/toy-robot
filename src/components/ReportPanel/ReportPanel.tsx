import React, { Component } from 'react';
import './styles/ReportPanel.css'
const uuidv4 = require('uuid/v4');

interface Props {
  results:Array<string>
}

export default class ReportPanel extends Component<Props> {
  constructor(props:Props) {
    super(props);
  }
  
  render() {
    return (
      <div id='report-panel-wrapper' className='report-panel-wrapper'>
        <span className='report-panel-label'>Results:</span>
        <div id='report-panel-results' className='report-panel-results'>
          {this.props.results.map((value,index)=> (<div key={uuidv4()}>{value}</div>))}
        </div>
      </div>
    );
  }
}