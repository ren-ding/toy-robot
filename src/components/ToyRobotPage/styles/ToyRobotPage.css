.toy-robot-page-wrapper
{
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items:center;
    margin: 40px;
}

.toy-robot-page-wrapper .toy-robot-page-header
{
    flex:1;
    font-size: 40px;
    line-height: 3;
    font-weight: bold;
}
  
.toy-robot-page-wrapper .toy-robot-page-container
{
    flex:6;
    display: flex;
    flex-direction: row;
}
  
.apply-button
{
    height: 30px;
    margin: auto;
}
  