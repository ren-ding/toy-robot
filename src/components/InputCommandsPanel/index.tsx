import InputCommandsPanel from './InputCommandsPanel'
export default InputCommandsPanel