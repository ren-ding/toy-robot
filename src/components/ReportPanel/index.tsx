import ReportPanel from './ReportPanel'
export default ReportPanel