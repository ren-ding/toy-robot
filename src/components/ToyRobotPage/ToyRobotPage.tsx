import React, { Component } from 'react';
import InputCommandsPanel from '../InputCommandsPanel/InputCommandsPanel'
import ReportPanel from '../ReportPanel/ReportPanel'

interface Props {}

interface State {
  inputCommands:string
  results:string
}

interface ApplyButtonProps {
  onClick:() => void
}

interface ContainerProps {
  results:string,
  onInputCommands:(event: {target: { value: string }}) => void,
  onApplyButtonClick:() => void
}

const Header = () => (
  <div>Toy-Robot Simulator</div>
);

const ApplyButton = ({
  onClick
}:ApplyButtonProps) => (
  <button type="button" id='apply-button' onClick={onClick}>apply</button>
);

const Container = ({
  results,
  onInputCommands,
  onApplyButtonClick
}:ContainerProps) =>(
  <div>
  <InputCommandsPanel 
    onInputCommands = {onInputCommands}
  />
  <ApplyButton
    onClick={onApplyButtonClick}
  />
  <ReportPanel results={results}/>
  </div>
);


export default class ToyRobotPage extends Component<Props,State> {
  constructor(props:Props) {
    super(props);
    this.state = {
      inputCommands: '',
      results:''
    };
  }

  render() {
    return (
      <div id='toy-robot-page-wrapper'>
        <Header />
        <Container 
          results = {this.state.results}
          onInputCommands = {this.onInputCommands}
          onApplyButtonClick = {this.onApplyButtonClick}
        />
      </div>
    );
  }

  onInputCommands = (event: { target: { value: string } }) => this.setState({inputCommands: event.target.value});

  //TODO:call the core algorithm to generate the report
  onApplyButtonClick = () => this.setState({results: this.state.inputCommands});
  
}
