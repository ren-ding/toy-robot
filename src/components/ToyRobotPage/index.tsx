import ToyRobotPage from './ToyRobotPage'
export default ToyRobotPage