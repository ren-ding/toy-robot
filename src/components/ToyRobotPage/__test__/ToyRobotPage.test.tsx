import React from 'react';
import ToyRobotPage from '../ToyRobotPage';
import InputCommandsPanel from '../../InputCommandsPanel/InputCommandsPanel';
import ReportPanel from '../../ReportPanel/ReportPanel';
import Enzyme, {mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

describe('ToyRobotPage', () => {
  let props:any;

  describe('render ', () => {
    it('should render to match snapshot', () => {
      const tree = renderer.create(<ToyRobotPage {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render body container include InputPanel, ReportPanel and an apply button', () => {
      const page = mount(<ToyRobotPage {...props}/>);
      expect(page.find(InputCommandsPanel).length).toBe(1);
      expect(page.find('ApplyButton').length).toBe(1);
      expect(page.find(ReportPanel).length).toBe(1);
    });
  });
});