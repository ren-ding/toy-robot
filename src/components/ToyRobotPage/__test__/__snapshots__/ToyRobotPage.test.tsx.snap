// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToyRobotPage render  should render to match snapshot 1`] = `
<div
  className="toy-robot-page-wrapper"
  id="toy-robot-page-wrapper"
>
  <div
    className="toy-robot-page-header"
    id="toy-robot-page-header"
  >
    Toy-Robot Simulator
  </div>
  <div
    className="toy-robot-page-container"
    id="toy-robot-page-container"
  >
    <div
      className="input-commands-panel-wrapper"
      id="input-commands-panel-wrapper"
    >
      <span
        className="input-label"
        id="input-label"
      >
        Please input commands
      </span>
      <textarea
        className="input-commands-textarea"
        onChange={[Function]}
        rows={20}
      />
    </div>
    <button
      className="apply-button"
      id="apply-button"
      onClick={[Function]}
      type="button"
    >
      apply
    </button>
    <div
      className="report-panel-wrapper"
      id="report-panel-wrapper"
    >
      <span
        className="report-panel-label"
      >
        Results:
      </span>
      <div
        className="report-panel-results"
        id="report-panel-results"
      />
    </div>
  </div>
</div>
`;
